test/mockserver/mockspanner.ts

```diff
@@ -0,0 +1,730 @@
+/*!
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {google} from '../../protos/protos';
+import * as grpc from 'grpc';
+import {ServiceError, status} from 'grpc';
+import * as protoLoader from '@grpc/proto-loader';
+import {Transaction} from '../../src';
+import protobuf = google.spanner.v1;
+import Timestamp = google.protobuf.Timestamp;
+import RetryInfo = google.rpc.RetryInfo;
+
+const PROTO_PATH = 'spanner.proto';
+const IMPORT_PATH = __dirname + '/../../../protos';
+const PROTO_DIR = __dirname + '/../../../protos/google/spanner/v1';
+
+/**
+ * Load the Spanner service proto.
+ */
+const packageDefinition = protoLoader.loadSync(PROTO_PATH, {
+  keepCase: false,
+  longs: String,
+  enums: String,
+  defaults: true,
+  oneofs: true,
+  includeDirs: [IMPORT_PATH, PROTO_DIR],
+});
+const protoDescriptor = grpc.loadPackageDefinition(packageDefinition);
+const spannerProtoDescriptor = protoDescriptor['google']['spanner']['v1'];
+const jsonProtos = require('../../../protos/protos.json');
+
+const RETRY_INFO = 'google.rpc.retryinfo-bin';
+// // tslint:disable-next-line variable-name
+// const RetryInfo = Root.fromJSON(jsonProtos).lookup('google.rpc.RetryInfo');
```

Comment:
I am not sure if we want to merge comment out code.

Reply:
Removed :-)